<launch>
  <!-- 在这里配置话题：机器人里程、参考轨迹、控制指令 -->
  <arg name="odom_topic" value="/state_ukf/odom" />
  <arg name="traj_topic" value="/trajectory" />
  <arg name="cmd_topic" value="cmd_vel" />

  <!-- mpc相关参数加载 -->
  <rosparam command="load" file="$(find mpc)/config/param.yaml" />
  
  <!-- mpc主程序 -->
  <node pkg="mpc" name="mpc_controller_node" type="mpc_controller_node" output="screen">
    <remap from="cmd_vel_"  to="$(arg cmd_topic)"/>
    <remap from="odom" to="$(arg odom_topic)"/>
    <remap from="traj" to="$(arg traj_topic)"/>
  </node>

</launch>
